### PR TITLE
20-10207 - Add Identity, Personal-info, & Living-situation chapters along with veteran-story e2e-spec

### PR DESCRIPTION
**Pull request by tlei123**
## Summary

- Add Your-identity, Your-personal-info, and Your-living-situation chapters to Priority Processing (20-10207), along with associated Veteran-story E2E-tests
- Veteran Facing Forms
- Flipper not implemented yet

## Related issue(s)

department-of-veterans-affairs/va.gov-team-forms#920
department-of-veterans-affairs/va.gov-team-forms#936

## Testing done

- Local browser-, unit-, & e2e-tests all successful

## What areas of the site does it impact?

This form ONLY

### CODE REVIEW

File: src/applications/simple-forms/20-10207/tests/e2e/veteran-story.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,371 @@
+/* eslint-disable @department-of-veterans-affairs/axe-check-required */
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/simple-forms/20-10207/tests/e2e/veteran-story.cypress.spec.js:1]: 10207-pp.cypress.spec.js [formTester] already axe-checks everything.  This add'l spec adds more coverage for negative-testing and custom-content.

File: src/platform/forms-system/src/js/web-component-patterns/fullNamePattern.js [hunk 2]
```diff
@@ -140,6 +148,13 @@ const fullNameSchema = commonDefinitions.fullName;
  */
 const fullNameNoSuffixSchema = commonDefinitions.fullNameNoSuffix;
 
+/**
+ * @returns `commonDefinitions.fullNameNoSuffix` minus middle name
+ */
+const noMiddleNoSuffixSchema = commonDefinitions.fullNameNoSuffix;
```
Comment (rhasselle-oddball) [inline src/platform/forms-system/src/js/web-component-patterns/fullNamePattern.js:154]: hm, this works. or what do you think of just adding firstAndLastNameUI, firstAndLastNameSchema, add not worrying about passing an omit property to fullNameSchema?

File: src/applications/simple-forms/20-10207/pages/otherHousingRisksThirdPartyVeteran.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import VaTextareaField from 'platform/forms-system/src/js/web-component-fields/VaTextareaField';
+import { titleUI } from 'platform/forms-system/src/js/web-component-patterns';
+
+import { ADDITIONAL_INFO_OTHER_HOUSING_RISKS_3RD_PTY_VET } from '../config/constants';
+
+/** @type {PageSchema} */
+export default {
+  uiSchema: {
+    ...titleUI('Other housing risks'),
+    'view:additionalInfoOtherHousingRisks': {
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/20-10207/pages/otherHousingRisksThirdPartyVeteran.js:10]: I wonder if you could just use either the second parameter of titleUI, or 'ui:description' here for all of these, which doesn't need a schema or name

File: src/applications/simple-forms/20-10207/pages/nameAndDateofBirth.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    // TODO: Use ...titleUI() once that supports functions for title
+    'ui:title': ({ formData }) => getNameAndDobPageTitle(formData),
+    fullName: fullNameNoSuffixUI(),
+    dateOfBirth: dateOfBirthUI({ required: true }),
+  },
+  schema: {
+    type: 'object',
+    properties: {
+      fullName: pdfFullNameNoSuffixSchema({
+        pdfMaxLengths: { first: 12, middle: 1, last: 18 },
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/20-10207/pages/nameAndDateofBirth.js:24]: what does the UI do for this? block input or give a character count warning? I wonder if design has recommendation

File: src/applications/simple-forms/20-10207/pages/idInfoThirdPartyVeteran.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  THIRD_PARTY_TYPES,
+  THIRD_PARTY_TYPE_VETERAN_LABELS,
+  ADDITIONAL_INFO_THIRD_PARTY_TYPE,
+} from '../config/constants';
+
+/** @type {PageSchema} */
+export default {
+  uiSchema: {
+    'ui:title': <h3 className="vads-u-margin-y--0">Your name</h3>,
+    thirdPartyFullName: fullNameNoSuffixUI(null, { omitMiddle: true }),
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/20-10207/pages/idInfoThirdPartyVeteran.js:20]: left comment below about maybe using a dedicated pattern for this

**Review by rhasselle-oddball — COMMENTED** [review]
👀 changes look good, test too, nice. some minor suggestions

**Comment by tlei123** [comment]
Closing this un-merged.  Got a bunch of merge conflicts trying to merge main into it.  Gonna create a new feature-branch and open a fresh PR.

File: src/platform/forms-system/src/js/web-component-patterns/fullNamePattern.js (hunk 2 above)
Comment (tlei123) [reply src/platform/forms-system/src/js/web-component-patterns/fullNamePattern.js:154]: Refactors done in new feature-branch.

File: src/applications/simple-forms/20-10207/pages/nameAndDateofBirth.js (hunk 4 above)
Comment (tlei123) [reply src/applications/simple-forms/20-10207/pages/nameAndDateofBirth.js:24]: Removed pdf max-lengths and associated code.

File: src/applications/simple-forms/20-10207/pages/idInfoThirdPartyVeteran.js (hunk 5 above)
Comment (tlei123) [reply src/applications/simple-forms/20-10207/pages/idInfoThirdPartyVeteran.js:20]: Now using ...titleUI(...)

File: src/applications/simple-forms/20-10207/pages/otherHousingRisksThirdPartyVeteran.js (hunk 3 above)
Comment (tlei123) [reply src/applications/simple-forms/20-10207/pages/otherHousingRisksThirdPartyVeteran.js:10]: Refactors done in new branch.

**Pull request closed without merging**